modules/pubsub/src/main/scala/zio/gcp/pubsub/topicadmin/TopicAdmin.scala

```diff
@@ -0,0 +1,114 @@
+package zio.gcp.pubsub.topicadmin
+
+import java.util.concurrent.TimeUnit
+import zio._
+import com.google.cloud.pubsub.v1.TopicAdminClient
+import com.google.cloud.pubsub.v1.stub.PublisherStub
+import com.google.cloud.pubsub.v1.TopicAdminSettings
+import com.google.pubsub.v1.DeleteTopicRequest
+import com.google.pubsub.v1.GetTopicRequest
+import com.google.pubsub.v1.ListTopicsRequest
+import com.google.pubsub.v1.ProjectName
+import com.google.pubsub.v1.ListTopicSubscriptionsRequest
+import com.google.pubsub.v1.UpdateTopicRequest
+import com.google.iam.v1.Policy
+import com.google.iam.v1.SetIamPolicyRequest
+import com.google.iam.v1.TestIamPermissionsResponse
+import com.google.iam.v1.TestIamPermissionsRequest
+import com.google.iam.v1.GetIamPolicyRequest
+import com.google.pubsub.v1.Topic
+
+object TopicAdmin {
+
+  trait Service {
+    def awaitTermination(duration: Long, unit: TimeUnit): Task[Boolean]
+    def createTopic(name: String): Task[Topic]
+    def createTopic(request: Topic): Task[Topic]
+    def deleteTopic(request: DeleteTopicRequest): Task[Unit]
+    def deleteTopic(topic: String): Task[Unit]
+    def getIamPolicy(request: com.google.iam.v1.GetIamPolicyRequest): Task[com.google.iam.v1.Policy]
+    def getSettings: Task[TopicAdminSettings]
+    def getStub: Task[PublisherStub]
+    def getTopic(request: GetTopicRequest): Task[Topic]
+    def getTopic(topic: String): Task[Topic]
+    def isShutdown(): Task[Boolean]
+    def listTopics(request: ListTopicsRequest): Task[TopicAdminClient.ListTopicsPagedResponse]
+    def listTopics(project: ProjectName): Task[TopicAdminClient.ListTopicsPagedResponse]
+    def listTopics(project: String): Task[TopicAdminClient.ListTopicsPagedResponse]
+    def listTopicSubscriptions(
+      request: ListTopicSubscriptionsRequest
+    ): Task[TopicAdminClient.ListTopicSubscriptionsPagedResponse]
+    def listTopicSubscriptions(topic: String): Task[TopicAdminClient.ListTopicSubscriptionsPagedResponse]
+    def setIamPolicy(request: SetIamPolicyRequest): Task[Policy]
+    def shutdown(): Task[Unit]
+    def shutdownNow(): Task[Unit]
```

Comment:
```suggestion
    def shutdownNow: Task[Unit]
```

Reply:
Also, does it make sense to have this in the service API?

Reply:
Yes. There is also a `close` method which I use to manage the releasing of the acquired `TopicAdmin` release. These `shutdown` methods are for the client to use to shutdown the topic in an orderly way, complete any outstanding work and prevent new messages to be published.